core/src/main/java/org/elasticsearch/index/translog/TranslogService.java

```diff
@@ -118,37 +124,64 @@ private TimeValue computeNextInterval() {
         return new TimeValue(interval.millis() + (ThreadLocalRandom.current().nextLong(interval.millis())));
     }
 
-    private class TranslogBasedFlush implements Runnable {
+    // public for testing
+    public TranslogBasedFlush createTranslogBasedFlush() {
+        return new TranslogBasedFlush();
+    }
+
+    public class TranslogBasedFlush extends AbstractRunnable {
 
         private volatile long lastFlushTime = System.currentTimeMillis();
 
```

Comment:
didn't really find a way to do this without changing too much in the semantics of the class and I'm also not sure how much it will buy us. Please ping to clarify, if need be.